pkg/deployment/inventory.go

```diff
@@ -139,7 +139,7 @@ func GenerateNodeInventory(ctx context.Context, helper *helper.Helper,
 	networkConfig := getAnsibleNetworkConfig(instance, instanceRole)
 
 	if networkConfig.Template != "" {
-		host.Vars["edpm_network_config_template"] = NicConfigTemplateFile
+		host.Vars["edpm_network_config_override"] = networkConfig.Template
```

Comment:
sorry I should have mentioned this eartlier, `edpm_network_config_override` is for static content (ansible vars not substituted) . If the customers are expected to use jinja templates inline in the spec, then this can't be used as we just use the `copy` module and not `template`.

Reply:
Eh, I should have checked. /sigh. 

In that case, I guess the current implementation is fine and doesn't need changing. I might just drop that commit from the list here and we can rescope this as a PR to add tests for the current implementation and add the necessary fixes to the envtest environment to facilitate that testing.